src/cider/nrepl/middleware/spec.clj

```diff
@@ -0,0 +1,128 @@
+(ns cider.nrepl.middleware.spec
+  (:require [cider.nrepl.middleware.util.error-handling :refer [with-safe-transport]]
+            [cider.nrepl.middleware.util.spec :as spec-utils]
+            [clojure.tools.nrepl
+             [middleware :refer [set-descriptor!]]
+             [misc :refer [response-for]]]
+            [cider.nrepl.middleware.util.cljs :as cljs]
+            [clojure.string :as str]
+            [clojure.walk :as walk]
+            [clojure.pprint :as pp]))
+
+(defn str-non-colls
+  "Given a form, convert all non collection childs to str"
+  [form]
+  (walk/postwalk #(if (coll? %)
+                    %
+                    (str %))
+                 form))
+
+(defn spec-list
+  "Retrieves a list of all specs in the registry, sorted by ns/name.
+  If filter-prefix is not empty, keep only the specs with that prefix."
+  [filter-prefix]
+  (let [sorted-specs (->> (spec-utils/registry)
+                          keys
+                          (map str) 
+                          sort)]
+    (if (not-empty filter-prefix)
+      (filter (fn [spec-symbol-str]
+                (let [checkable-part (if (.startsWith spec-symbol-str ":")
+                                       (subs spec-symbol-str 1)
+                                       spec-symbol-str)]
+                 (.startsWith checkable-part filter-prefix)))
+              sorted-specs)
+      sorted-specs)))
+
+
+(defn get-multi-spec-sub-specs
+  "Given a multi-spec form, call it's multi method methods to retrieve
+  it's subspecs"
+  [multi-spec-form]
+  (let [[_ mm-symbol & _] multi-spec-form]
+   (->> @(resolve mm-symbol)
+        methods 
+        (map (fn [[spec-k method]]
+               [spec-k (spec-utils/form (method nil))])))))
+
+(defn add-multi-specs
+  "Walk down a spec form and for every subform that is a multi-spec
+  add it's sub specs"
+  [form]
+  (walk/postwalk (fn [sub-form]
+                   (if (and (coll? sub-form)
+                            (symbol? (first sub-form))
+                            (-> sub-form first name (= "multi-spec")))
+                     (concat sub-form (get-multi-spec-sub-specs sub-form))
+                     sub-form))
+                 form))
+
+(defn spec-from-string
+  "Given a string like `clojure.core/let` or `:user/email` returns
```

Comment:
I'd put string examples in quotes, not backticks (those are typically reserved for references).